docs/Explanation/ADR/adr-004-did.md

```diff
@@ -0,0 +1,687 @@
+
+- `created`: a [datetime](https://www.w3.org/TR/xmlschema11-2/#dateTime) string of the creation date, that is the utc date associated to the block height when the DID document was submitted the first time
+- `updated`: a [datetime](https://www.w3.org/TR/xmlschema11-2/#dateTime) string of the last update date, that is the utc date associated to the block height when the DID document was submitted the last time
+- `deactivated`: a boolean field that indicates whenever the DID document is [deactivated](#Deactivate) 
+- `versionId`: for the version id we use the hex encoded blake2b hash of the transaction that created/updated the DID
+
+##### [DID Resolution Metadata](https://www.w3.org/TR/did-core/#did-resolution-metadata)
+
+The resolution metadata are outside the scope of the gRPC interface and are not covered in this ADR
+
+#### [DID URL Syntax](https://www.w3.org/TR/did-core/#did-url-syntax)
+
+The DID URL is outside the scope of the gRPC interface and are not covered in this ADR
+##### [Query parameters](https://www.w3.org/TR/did-core/#did-parameters)
+
+The query parameters URL is outside the scope of the gRPC interface and are not covered in this ADR
+
+<!-- 
+
+The implementation MUST support the following query parameters:
+
+- `versionId` - to retrieve a DID document with a specific version 
+- `versionTime` - to retrieve the version of the DID document valid at a specific time, the parameter MUST be a valid [datetime](https://www.w3.org/TR/xmlschema11-2/#dateTime).   
+
+The format for the queries is:
+-->
+## Consequences
+
+The cosmos ecosystem will have at its disposition a DID module compatible with the W3C standard and with a high chance of compatibility with 3rd party components such as cloud and edge agents, resolvers and so on.
+
+### Backwards Compatibility
+
+This is a new module so backward compatibility is not a concern.
+
+### Positive
+
+- The implementation of the ADR provides the foundation for interoperability with the DID standard and more in general with SSI identity approach
+- Closely following the W3C standard gives the best chances of successful interoperability with 3rd party components.
+
+### Negative
+
+- The implementation follows rigidly the W3C specification leaving little room for extensibility, that might became an issue for a wider adoption 
+
+### Neutral
+
+N/A
+
+## Further Discussions
+
+While an ADR is in the DRAFT or PROPOSED stage, this section contains a summary of issues to be solved in future iterations. The issues summarized here can reference comments from a pull request discussion.
+Later, this section can optionally list ideas or improvements the author or reviewers found during the analysis of this ADR.
+
+- The `did:key` method specifies a key format that is not the one used in this ADR. The ADR needs to be amended or to follow a different approach
+- The approach proposed is somewhat locked in to the current implementation, it will have to be revised in successive iterations. 
```

Comment:
```suggestion
- The `did:key` method specifies a key format that is different from the one used in this ADR. This ADR needs to be amended or follow a different approach.
- The approach proposed is somewhat locked into the current implementation and will have to be revised in successive iterations. 
```

Reply:
hmmm which `did:key` method are we talking about in line 673? if we mean the `did:key` method  in the Cosmos SDK, let's say that.